### PR TITLE
Onyx username

### PR DESCRIPTION
**Pull request by anas-shami**
Docs Pull Request
Feature Pull Request
New Module Pull Request
Support onyx username commands:
- username *
- username * password *
- username * nopassword
- username * disable login
- username * capability admin
- no username *
- no username * disable

which able to change the user configurations or just create a new one.

### CODE REVIEW

File: lib/ansible/modules/network/onyx/onyx_username.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        """
+        module initialization
+        """
+        element_spec = dict()
+
+        argument_spec = dict(state=dict(choices=['absent', 'present'], default='present'),
+                             username=dict(type='str', required=True),
+                             disabled=dict(choices=['none', 'login', 'password', 'all']),
+                             capability=dict(choices=['monitor', 'unpriv', 'v_admin', 'admin']),
+                             nopassword=dict(type='bool', default=False),
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_username.py:147]: password and nopassword should be mutual exclusive

**Review by samerd — CHANGES_REQUESTED** [review]
Thanks @anas-shami 
just a small comment, please fix

File: lib/ansible/modules/network/onyx/onyx_username.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+                             password=dict(type='str'),
```
Comment (ganeshrn) [inline lib/ansible/modules/network/onyx/onyx_username.py]: Add `no_log=True` in argspec for password related options

Comment (anas-shami) [reply lib/ansible/modules/network/onyx/onyx_username.py]: I'll add it 
Thanks,

**Comment by anas-shami** [comment]
@samerd please review

**Review by samerd — APPROVED** [review]
looks great
thanks @anas-shami

**Comment by samerd** [comment]
thanks @ganeshrn for your review.
could you please approve now?

**Comment by anas-shami** [comment]
@ganeshrn @samerd please approve I remove debugging line that causes this fail

**Pull request merged**